Client/Frontend/Home/MessageCard/MessageCardDataAdaptor.swift

```diff
@@ -0,0 +1,53 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+protocol MessageCardDataAdaptor {
+    func getMessageCardData() -> GleanPlumbMessage?
+}
+
+protocol MessageCardDelegate: AnyObject {
+    func didLoadNewData()
+}
+
+class MessageCardDataAdaptorImplementation: MessageCardDataAdaptor, GleanPlumbMessageManagable {
+
+    weak var delegate: MessageCardDelegate?
+    private var message: GleanPlumbMessage?
+    var notificationCenter: NotificationProtocol
+
+    func getMessageCardData() -> GleanPlumbMessage? {
+        return message
+    }
+
+    init(notificationCenter: NotificationProtocol = NotificationCenter.default) {
+        self.notificationCenter = notificationCenter
+        setupNotifications(forObserver: self, observing: [UIApplication.willEnterForegroundNotification])
+
+        updateMessage()
+    }
+
+    /// Call messagingManager to retrieve the message for new tab card
+    /// An expired message will not trigger a reload of the section
+    /// - Parameter surface: Message surface id
+    private func updateMessage(for surface: MessageSurfaceId = .newTabCard) {
+        guard let validMessage = messagingManager.getNextMessage(for: surface) else { return }
+
+        if !validMessage.isExpired {
+            message = validMessage
+            delegate?.didLoadNewData()
+        }
+    }
+}
+
+extension MessageCardDataAdaptorImplementation: Notifiable {
+    func handleNotifications(_ notification: Notification) {
```

Comment:
I'm not sure if this is the desired behaviour for messaging because the message shouldn't change during the app launch. Now we only have one message for that surface so we will always get the same one but I wonder if we have multiple it might change every time we enter foreground